ios/BluedotPointSDK.m

```diff
@@ -30,26 +34,130 @@ - (instancetype)init {
     return self;
 }
 
-RCT_EXPORT_METHOD(authenticate:(NSString *)projectId
-    requestAuthorization:(NSString *)authorizationLevel
-    authenticationSuccessful:(RCTResponseSenderBlock)authenticationSuccessfulCallback
-    authenticationFailed: (RCTResponseSenderBlock)authenticationFailedCallback)
+RCT_EXPORT_METHOD(initialize:(NSString *)projectId
+    initializationSuccessful:(RCTResponseSenderBlock)initializationSuccessfulCallback
+    initializationFailed: (RCTResponseSenderBlock)initializationFailedCallback)
+{
+    [[BDLocationManager instance] initializeWithProjectId: projectId completion:^(NSError * error)
+     {
+         if (error != nil) {
+             initializationFailedCallback(@[error.localizedDescription]);
+         } else {
+             initializationSuccessfulCallback(@[]);
+         }
+     }];
+}
+
+RCT_REMAP_METHOD(isInitialized,
+                 isInitializedResolver: (RCTPromiseResolveBlock)resolve
+                 isInitializedRejecter: (RCTPromiseRejectBlock)reject)
+{
+    BOOL isInitialized = [ BDLocationManager.instance isInitialized ];
+    NSNumber *output = [NSNumber numberWithBool: isInitialized ];
+
+    resolve(output);
+}
+
+RCT_EXPORT_METHOD(reset:(RCTResponseSenderBlock)resetSuccessfulCallback
+    resetFailed:(RCTResponseSenderBlock)resetFailedCallback)
+{
+    [[BDLocationManager instance] resetWithCompletion:^(NSError * error)
+    {
+        if (error != nil) {
+            resetFailedCallback(@[error.localizedDescription]);
+        } else {
+            resetSuccessfulCallback(@[]);
+        }
+    }];
+}
+
+RCT_EXPORT_METHOD(iOSStartGeotriggering:(RCTResponseSenderBlock)startGeotriggeringSuccessfulCallback
+                  startGeotriggeringFailed:(RCTResponseSenderBlock)startGeotriggeringFailedCallback)
 {
-    BDAuthorizationLevel bdAuthorizationLevel;
     
-    if ([authorizationLevel isEqualToString:@"WhenInUse"])
+    [[BDLocationManager instance] startGeoTriggeringWithCompletion:^(NSError * error)
     {
-        bdAuthorizationLevel = authorizedWhenInUse;
-    } else {
-        bdAuthorizationLevel = authorizedAlways;
-    }
-        
-    _callbackAuthenticationSuccessful = authenticationSuccessfulCallback;
```

Comment:
@toroduque you can also remove the local variable from line 8-11 since they are no longer being used:
`RCTResponseSenderBlock _callbackAuthenticationSuccessful;
    RCTResponseSenderBlock _callbackAuthenticationFailed;
    RCTResponseSenderBlock _callbackLogOutSuccessful;
    RCTResponseSenderBlock _callbackLogOutFailed;`

Reply:
Removed